### PR TITLE
update from 1.0 to 2.0

### PR DESCRIPTION
**Pull request by Walterm21**

### CODE REVIEW

**Comment by Walterm21** [comment]
fixed the functionality issue.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+version: 2
+jobs:
+  build:
```
Comment (joshua-rutherford) [inline .circleci/config.yml]: As it stands this job doesn't build.  It should have a step after `- checkout` that is something like `- run: mvn clean compile`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - image: circleci/openjdk:8
+    steps:
+      - checkout
+  test:
+    docker:
+      - image: circleci/openjdk:8
+    steps:
+      - checkout
+      - run: mvn verify
+      - run: if [ -z "${CI_PULL_REQUEST}" ]; then PGP_RINGS="$(echo ${PGP_RINGS_ENCODED} | base64 --decode)" mvn deploy -P deploy -s settings.xml; fi
```
Comment (joshua-rutherford) [inline .circleci/config.yml]: This run statement actually deploys, yet the job is called `test`.  I think you should make a third `job` named `deploy` and move this into it.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+dependencies:
+    pre:
+        - mvn dependency:go-offline
+workflows:
+  version: 2
+  test-and-deploy:
+    jobs:
+      - test
+      - deploy:
```
Comment (joshua-rutherford) [inline .circleci/config.yml]: This can't work as you don't have a job titled `deploy` at all.

**Comment by joshua-rutherford** [comment]
This is now OBE.

**Pull request closed without merging**